src/dex-ui/pages/ProposalDetails/ProposalDetailsControls.tsx

```diff
@@ -143,7 +154,39 @@ export function ProposalDetailsControls(props: ProposalDetailsControlsProps) {
     );
   }
 
-  if (areVoteButtonsVisible) {
+  if (areVoteButtonsVisible && !isWalletConnected) {
+    return (
+      <>
+        <Text textStyle="h3">Vote on Proposal</Text>
+        <Flex gap="4"></Flex>
+        <Notification
+          type={NotficationTypes.WARNING}
+          message={"You’ll need to connect a wallet before you can vote on a proposal."}
+        />
+      </>
+    );
+  }
+
+  if (areVoteButtonsVisible && isWalletConnected && !doesUserHaveGodTokens) {
+    return (
+      <>
+        <Text textStyle="h3">Vote on Proposal</Text>
+        <Flex direction="column" gap="4"></Flex>
```

Comment:
Flexbox (`Flex`) is intended to determine element layouts. If you just need spacing, you can use the `Spacer` component from ChakraUI. `margin` or `padding` can be added to the `Spacer` component.

Reply:
If you end up wrapping the `Text` and `Notification` elements in a `Flex` element, you may be able to set the `gap` attribute instead of using a `Spacer` component to create vertical spacing between the elements.

Reply:
My bad, I did it that way to keep UI elements under `Flex` tag but somehow code completion messed up and yeah it does not make any sense to use empty `Flex` tags just for spacing.
I have corrected it to right way now.